src/mlj_interface.jl

```diff
@@ -143,21 +139,36 @@ function MMI.transform(m::KMeans, fitresult, Xnew)
     end
 
     # Warn users if fitresult is from a `non-converged` fit
-    if !fitresult[end].converged
+    if !(fitresult.converged)
         @warn "Failed to converge. Using last assignments to make transformations."
     end
 
-    # results from fitted model
-    results = fitresult[1]
-
     # use centroid matrix to assign clusters for new data
-    centroids = results.centers
-    distances = Distances.pairwise(Distances.SqEuclidean(), DMatrix, centroids; dims=2)
-    preds = argmin.(eachrow(distances))
-    return MMI.table(reshape(preds, :, 1), prototype=Xnew)
+    distances = Distances.pairwise(Distances.SqEuclidean(), DMatrix, fitresult.centers; dims=2)
```

Comment:
I am not sure about `SqEuclidean` here. 
Despite the fact that internally most of the time we operate with the squared norm (for performance), our final clusters are calculated in the usual `Euclidean` metric.

I think, we should switch here.

Reply:
Switching.